src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -347,6 +351,48 @@
     @QueryParam("disableRemainingWeightHeuristic")
     protected Boolean disableRemainingWeightHeuristic;
 
+    /**
+     * Extra penalty added for flag stop boarding/alighting.
+     */
+    @QueryParam("flagStopExtraPenalty")
+    protected Integer flagStopExtraPenalty;
+
+    /**
+     * Extra penalty added for deviated-route boarding/alighting.
+     */
+    @QueryParam("deviatedRouteExtraPenalty")
+    protected Integer deviatedRouteExtraPenalty;
+
+    /**
+     * How much worse is call-n-ride than regular transit (multiplier for weight)
+     */
+    @QueryParam("callAndRideReluctance")
+    protected Double callAndRideReluctance;
+
+    /*
+     * Size of flag stop buffer zone in UI
+     */
+    @QueryParam("flagStopBufferSize")
+    protected Double flagStopBufferSize;
```

Comment:
OTP computes a specific point for flag stop board/alights, but the UI designs are to show an _area_ along the street segment where the bus can be boarded or alighted from. We want to make it clear to users that the bus can be boarded or alighted from anywhere, and OTP's optimized point may not be the best in reality (e.g. no place for a customer to stand). Since the UI doesn't have street network data, OTP has to provide the buffer segment around the computed point. This is a configuration parameter for how large the buffer should be. I'll add more documentation...

![screen shot 2018-09-21 at 4 25 05 pm](https://user-images.githubusercontent.com/5751667/45904515-4d681180-bdbb-11e8-87de-f75550936db6.png)

Reply:
I'm somewhat hesitant about passing rendering-related information into the routing system, in order to modify the result so it will render in a particular way on a particular UI. I guess it's analogous to passing in a metric/imperial units parameter and having the back end make the conversion. I am generally more in favor of having a standard API response and having the front end transform it as needed for display in a particular environment.

This seems like another thing that might not really be suited to be a per-request parameter, but rather a server configuration option in JSON.

Reply:
It is also important to include the units (in the parameter name itself if possible).